### PR TITLE
mc: dont add libX11.so on darwin

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/131002#discussion_r676794610
Thanks @bobrik 
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/mc/default.nix [hunk 1]
```diff
@@ -60,7 +60,7 @@ stdenv.mkDerivation rec {
   postFixup = ''
     # remove unwanted build-dependency references
     sed -i -e "s!PKG_CONFIG_PATH=''${PKG_CONFIG_PATH}!PKG_CONFIG_PATH=$(echo "$PKG_CONFIG_PATH" | sed -e 's/./0/g')!" $out/bin/mc
-
+  '' + lib.optionalString (!stdenv.isDarwin) ''
```
Comment (bobrik) [inline pkgs/tools/misc/mc/default.nix:63]: I don't know if it should be this or `stdenv.isLinux`, maybe somebody else could chime in.

**Review by bobrik — APPROVED** [review]

File: pkgs/tools/misc/mc/default.nix (hunk 1 above)
Comment (svanderburg) [reply pkgs/tools/misc/mc/default.nix:63]: Sounds like a good approach. On Linux (and other UNIX-like systems as well, e.g. FreeBSD), we typically have X11 at our disposal as a display manager. On macOS, we only need an X display server in very rare cases.

Comment (Artturin) [reply pkgs/tools/misc/mc/default.nix:63]: the buildInputs has
```
  ] ++ lib.optionals (!stdenv.isDarwin) [ e2fsprogs gpm ];``` 
	so i followed that

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 131592` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mc</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 131592` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mc</li>
  </ul>
</details>

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 131592` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mc</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Comment by figsoda** [comment]
misclicked, my bad

**Pull request merged**